friendship/models.py

```diff
@@ -341,8 +331,8 @@ def add_friend(self, from_user, to_user, message=None):
     def remove_friend(self, from_user, to_user):
         """ Destroy a friendship relationship """
         try:
-            qs = Friend.objects.filter(Q(to_user=to_user, from_user=from_user) | Q(to_user=from_user, from_user=to_user))
-            distinct_qs = qs.distinct().all()
+            qs = Friend.objects.filter(to_user__in=[to_user, from_user], from_user__in=[from_user, to_user])
```

Comment:
Could be both to_user or from_user.

Reply:
I prefer `qs = Friend.objects.filter(Q(to_user=to_user, from_user=from_user) | Q(to_user=from_user, from_user=to_user))`.

Reply:
I'm not thrilled by it either, but it solves several edge cases that users can find themselves unknowingly in with update() or any of the bulk_*() methods. Ideally, we should add a Constraint too, but I was apprehensive about adding that migration yet since it can cause existing users headaches.